File: python3/LeetCode/totalHammingDistance.py
from typing import List


class Solution:
    def totalHammingDistance(self, nums:List[int]) -> int:
        res = 0
        n = len(nums)
        for i in range(0,31):
            temp =0
            for num in nums:
                temp+=(num>>i)&1
            res+= temp*(n-temp)
        return res

if __name__ == '__main__':
    nums = [4,14,2]
    print(Solution().totalHammingDistance(nums))

File: go/LeetCode/totalHammingDistance.go
package main

import "fmt"

func main() {
	nums := []int{4,14,2}
	fmt.Print(totalHammingDistance(nums))
}

func totalHammingDistance(nums []int) int {
	n:=len(nums)
	res:=0
	for i:=0;i<31;i++ {
		temp:=0
		for _,v := range nums{
			temp+=(v>>i)&1
		}
		res+=temp*(n-temp)
	}
	return res
}
